pkg/queue/instance.go

```diff
@@ -46,14 +47,16 @@ type Instance interface {
 
 type queueImpl struct {
 	delay     time.Duration
-	tasks     []Task
+	tasks     []*Task
 	cond      *sync.Cond
 	closing   bool
 	closed    chan struct{}
 	closeOnce *sync.Once
 	// initialSync indicates the queue has initially "synced".
-	initialSync *atomic.Bool
-	id          string
+	initialSync   *atomic.Bool
+	id            string
+	metrics       *queueMetrics
+	enableMetrics bool
```

Comment:
very nit: may be simpler to just make `metrics` nil if EnableControllerQueueMetrics==false